### DISCUSSION TITLE
I suddenly got this issue in Visual Studio that I have not got before, 'irccommand.h' No such file or directory

### DISCUSSION THREAD

**Discussion by karl-police** [Chatterino/chatterino2#3492] [Q&A] [answered]
I setup the things according to https://github.com/Chatterino/chatterino2/blob/master/BUILDING_ON_WINDOWS.md

However, instead of making a build in QtCreator I was able to successfully make a build with Visual Studio, by importing it with a Qt Tools extension.

Now, then I closed Visual Studio with Task Manager, because it had some issue.

Then encountered an issue.

Then I repaired the installation of Visual Studio.

And now I get this issue "fatal error C1083: Cannot open include file: 'irccommand.h': No such file or directory". However, things that are something like ``#include <QCheckBox>`` do not make any issues.

What have I done wrong? Where is ``irccommand.h`` from and how can I get, and how does it come that I did not encounter any issues before, but now I do?

Is ``irccommand.h`` from here? http://communi.github.io/

**Comment by karl-police** [comment]
Things like ``<pajlada/serialize.hpp>`` could also not be opened, not even sure where that is supposed to be at.

**Comment by Mm2PL** [comment, marked as answer]
You forgot to fetch git submodules, you can do so with `git submodule update --init --recursive`

**Reply by karl-police** [reply]
> You forgot to fetch git submodules, you can do so with `git submodule update --init --recursive`

I actually used Github Desktop to fetch the repository, but not under Chatterino, the one that I forked.


Is there where the issue came from?

Where would I run that command?

**Reply by karl-police** [reply]
> You forgot to fetch git submodules, you can do so with `git submodule update --init --recursive`

Also did I miss something that I forgot about "git submodule" ?

**Reply by Mm2PL** [reply]
It doesn't matter which repo you pulled  (upstream or your fork), you are missing libraries required to compile the thing. You would type that into the terminal/command line. Github desktop probably has some menu somewhere that has a Submodule Update operation, at least I would imagine it having it. See [git's manual](https://git-scm.com/docs/git-submodule) or [the book](https://git-scm.com/book/en/v2/Git-Tools-Submodules) for more info on submodules.

**Reply by karl-police** [reply]
> You forgot to fetch git submodules, you can do so with `git submodule update --init --recursive`

I was able to identify how this issue happend and therefore how to solve it.

It was related to "submodules" as you mentioned, not exactly sure what they are, but maybe related to the reason why you can't find files that are in a forked repository through search, except if the file from the forked repository is from the forked repository and not from the repository it was forked from.
Github Desktop handles that when a repository gets cloned. However, I manually deleted my cloned repository and switched branches because I thought I would "clean" the folder from wrong files, but that didn't seem so.

So I had to just clone the repository again.

My plan is that I clone the repository, copy the cloned files and put them in another folder so that I can import that into Visual Studio. And then Visual Studio creates the project files and stuff inside of it, and messes up with pull requests and stuff.

Not sure if there would be a better way of doing that.
